include/swift/SIL/SILBuilder.h

```diff
@@ -526,8 +575,11 @@ class SILBuilder {
 
   UncheckedRefCastInst *createUncheckedRefCast(SILLocation Loc, SILValue Op,
                                                SILType Ty) {
-    return insert(new (F.getModule()) UncheckedRefCastInst(
+    auto *I = insert(new (F.getModule()) UncheckedRefCastInst(
         getSILDebugLocation(Loc), Op, Ty));
+    I->getTypeDefOperandsBuf()[0].set(
+        getOpenedArchetypeDef(Ty, getModule(), OpenedArchetypes));
+    return I;
```

Comment:
@rjmccall The minor annoyance with this approach is that a number of affected instructions are currently derived from `UnaryInstructionBase` and then extend it with more class members. But this would not work anymore if the number of "open archetype def" operands is not statically known, because now they should be tail allocated, which means that no further class members should follow them.  I guess these instructions need to be derived directly from the `SILInstruction` now.